### DISCUSSION TITLE
Why in evaluation there are 1856 predictions

### DISCUSSION THREAD

**Discussion by fersebas** [PaddlePaddle/PaddleSpatial#30] [BaiduKDDCup2022] [answered]
Why there are 1856 predictions  in evaluation  (last 15 days) instead of 15*24*6 - 288 + 1 = 1873? It depends on batch size?

Thank you for your answer.

Fernando

**Comment by DrownFish19** [comment, marked as answer]
The dataloader with `drop_last =True` will drop the last items which can not make another batch.

In the baseline, the batchsize is set at 32, and 1873/32 = 58.53125, 1856/32 = 58. Thus the last 17 items will be drop.

```
        if flag == 'test':
            shuffle_flag = False
            drop_last = True
        else:
            shuffle_flag = True
            drop_last = True

        ......
        data_loader = DataLoader(
            data_set,
            batch_size=self.args["batch_size"],
            shuffle=shuffle_flag,
            num_workers=self.args["num_workers"],
            drop_last=drop_last
        )
```

**Reply by ramber1836** [reply]
The answer of DrownFish19 is correct. Thanks, DrownFish19. 

For the 48-hour-prediction rolling on the (15 days) test set with stride 1, there is supposed to be 24 * 6 * 15 - 288 + 1= 1873 predictions. The baseline code conducts the prediction in a batch style. If drop_last is True, the evaluated predictions could be less than the real ones.

This should be avoided in the future-released evaluation code for fair competition.

**Reply by bingblackbean** [reply]
when the updated evaluation code will be released? It seems the batch-size matters in the current approach, which is not a good practice.

**Reply by ramber1836** [reply]
The updated code will be released in a couple of days.

**Reply by zhoujingbo** [reply]
> when the updated evaluation code will be released? It seems the batch-size matters in the current approach, which is not a good practice.

Hi, this is to update that we have updated the baseline

**Reply by phoenixes94** [reply]
def __read_data__(self):
        self.scaler = Scaler()
        df_raw = pd.read_csv(os.path.join(self.data_path, self.filename))
        border1s = [self.tid * self.total_size,
                    self.tid * self.total_size + self.train_size - self.input_len,
                    self.tid * self.total_size + self.train_size + self.val_size - self.input_len
                    ]
        border2s = [self.tid * self.total_size + self.train_size,
                    self.tid * self.total_size + self.train_size + self.val_size,
                    self.tid * self.total_size + self.train_size + self.val_size + self.test_size
                    ]
        border1 = border1s[self.set_type]
        border2 = border2s[self.set_type]

        df_data = df_raw
        if self.task == 'M':
            cols_data = df_raw.columns[self.start_col:]
            df_data = df_raw[cols_data]
        elif self.task == 'MS':
            cols_data = df_raw.columns[self.start_col:]
            df_data = df_raw[cols_data]
        elif self.task == 'S':
            df_data = df_raw[[self.tid, self.target]]

        # Turn off the SettingWithCopyWarning
        pd.set_option('mode.chained_assignment', None)
        df_data.replace(to_replace=np.nan, value=0, inplace=True)

        if self.scale:
            train_data = df_data[border1s[0]:border2s[0]]
            self.scaler.fit(train_data.values)
            data = self.scaler.transform(df_data.values)
        else:
            data = df_data.values
        self.data_x = data[border1:border2]
        self.data_y = data[border1:border2]
        self.raw_data = df_data[border1 + self.input_len:border2]
        
Self.tid * self.total_size + self.train_size + self.val_size - self.input_len Self. 
data_x has the same dimension as self.data_y (16* 144,10), and self.raw_data (15* 144,10). Shouldn't the latitude of the predictions be int((24 * 6 * 16-288 + 1)/32)*32=2016?
![1658128490(1)](https://user-images.githubusercontent.com/54425898/179461292-3f8e39b1-9bd3-4918-9860-c9f9239f81f5.png)

**Reply by phoenixes94** [reply]
Looking forward to your reply!

**Reply by ramber1836** [reply]
As we mentioned before, we will not answer any questions related to the technical details in the Phase 3 competition. 
Sorry for that.

**Reply by phoenixes94** [reply]
I just want to understand how the baseline code works, which is not a technical question about the structure of the model. Such problems in the past seem to be explained.
